packages/lumx-react/src/components/tabs/Tab.tsx

```diff
@@ -39,45 +36,80 @@ const CLASSNAME = `${CSS_PREFIX}-tabs__link`;
  */
 const DEFAULT_PROPS: Partial<TabProps> = {};
 
-const Tab: React.FC<TabProps> = ({
-    className,
-    disabled,
-    icon,
-    index,
-    isActive,
-    isDisabled = disabled,
-    label,
-    onTabClick,
-    ...forwardedProps
-}) => {
-    const handleTabClick: MouseEventHandler = (event) => {
-        onTabClick?.({ event, index });
-    };
+/**
+ * Tab component.
+ *
+ * Implements WAI-ARIA `tab` role {@see https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html#rps_label}
+ *
+ * @param  props Component props.
+ * @return React element.
+ */
+const Tab: React.FC<TabProps> = (props) => {
+    const {
+        className,
+        disabled,
+        icon,
+        id,
+        isActive: propIsActive,
+        isDisabled = disabled,
+        label,
+        onFocus,
+        onKeyPress,
+        tabIndex = -1,
+        ...forwardedProps
+    } = props;
+    const context = TabProvider.useContext();
+    const { index, tabId, tabPanelId, isActive: stateIsActive } = context.useTab(id);
+    const isActive = propIsActive || stateIsActive;
 
-    const handleKeyPress: KeyboardEventHandler = (event) => {
-        const keyCode = event.which ?? event.keyCode;
+    const handleFocus: FocusEventHandler = useCallback(
+        (event) => {
+            onFocus?.(event);
+            if (context.shouldActivateOnFocus) {
+                context.onChange(index);
+            }
+        },
+        [index, onFocus, context],
+    );
 
-        if (keyCode !== ENTER_KEY_CODE) {
-            return;
-        }
+    const handleClick: MouseEventHandler = useCallback(() => {
+        context.onChange(index);
+    }, [context, index]);
 
-        onTabClick?.({ event, index });
-    };
+    const handleKeyPress: KeyboardEventHandler = useCallback(
+        (event) => {
+            onKeyPress?.(event);
+            const keyCode = event.which ?? event.keyCode;
+            if (keyCode !== ENTER_KEY_CODE) {
+                return;
+            }
+
+            context.onChange(index);
+        },
+        [index, onKeyPress, context],
+    );
 
     return (
-        <a
+        <button
             {...forwardedProps}
+            id={tabId}
             className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, isActive, isDisabled }))}
-            tabIndex={isDisabled ? -1 : 0}
-            onClick={handleTabClick}
+            onClick={handleClick}
             onKeyPress={handleKeyPress}
+            onFocus={handleFocus}
+            role="tab"
+            tabIndex={isActive ? 0 : tabIndex}
+            disabled={isDisabled}
```

Comment:
As noted in another comment (https://github.com/lumapps/design-system/pull/540/files#r527516518), disabled tabs need to still be focusable (but not clickable).
This is why only the aria-disabled was used + click was handled to not trigger when tab was disabled.

Adding the "disabled" attribute completely blocks the focus from being set on the tab...

Reply:
removing